pkg/providers/instance/instance.go

```diff
@@ -224,6 +224,9 @@ func (p *DefaultProvider) launchInstance(ctx context.Context, nodeClass *v1beta1
 		},
 	}
 	if capacityType == corev1beta1.CapacityTypeSpot {
+		if options.FromContext(ctx).InterruptionQueue == "" {
```

Comment:
I don't think that we should be gating the use of spot based on if someone has the interruption queue enabled or not. I'd imagine if we wanted to do this, that we would just require the interruption queue to be set outright since part of Karpenter's value proposition is that it makes it so easy to use spot out of the box. 

Additionally, you can run spot without using the interruption queue (not sure I'd recommend it, but you definitely can, and some people use NTH (Node Termination Handler) rather than using Karpenter's in-built interruption handling.

Reply:
Bear in mind that this is just a Warn message. I'll remove this from the PR for now. Thanks for the quick feedback, really appreciate